src/lib/types.ts

```diff
@@ -209,14 +210,335 @@ type ResolveInviteCodeResponse = {
+  type: "calendar_created";
+  data: {
+    fields: CalendarFields;
+  };
+};
+
+type CalendarUpdated = {
+  type: "calendar_updated";
+  data: {
+    id: Hash;
+    fields: CalendarFields;
+  };
+};
+
+type SpacesPageDescriptionUpdated = {
+  type: "space_page_description_updated";
+  data: {
+    description: string;
+  };
+};
+
+type ResourcesPageDescriptionUpdated = {
+  type: "space_page_description_updated";
+  data: {
+    description: string;
+  };
+};
+
+type AboutPageDescriptionUpdated = {
+  type: "space_page_description_updated";
+  data: {
+    description: string;
+  };
+};
+
+type CalendarDeleted = {
+  type: "calendar_deleted";
+};
+
+/**
+ * Space
+ */
+
+type SpaceCreated = {
+  type: "space_created";
+  data: {
+    field: SpaceFields;
+  };
+};
+
+type SpaceUpdated = {
+  type: "space_updated";
+  data: {
+    id: Hash;
+    fields: SpaceFields;
+  };
+};
+
+type SpaceDeleted = {
+  type: "space_deleted";
+};
+
+/**
+ * Resource
+ */
+
+type ResourceCreated = {
+  type: "resource_created";
+  data: {
+    fields: ResourceFields;
+  };
+};
+
+type ResourceUpdated = {
+  type: "space_updated";
+  data: {
+    id: Hash;
+    fields: ResourceFields;
+  };
+};
+
+type ResourceDeleted = {
+  type: "resource_deleted";
+};
+
+/**
+ * Event
+ */
+
+type EventCreated = {
+  type: "event_created";
+  data: {
+    fields: EventFields;
+  };
+};
+
+type EventUpdated = {
+  type: "event_updated";
+  data: {
+    id: Hash;
+    fields: EventFields;
+  };
+};
+
+type EventDeleted = {
+  type: "event_deleted";
+  data: {
+    id: Hash;
+  };
+};
+
+/**
+ * Requests and responses
+ */
+
+type SpaceRequested = {
+  type: "space_requested";
+  data: {
+    eventId: Hash;
+    spaceId: Hash;
+    message: string;
+    timePeriod: TimeSpan[];
+  };
+};
+
+type SpaceRequestAccepted = {
+  type: "space_request_accepted";
+  data: {
+    requestId: Hash;
+  };
+};
+
+type SpaceRequestRejected = {
+  type: "space_request_rejected";
+  data: {
+    requestId: Hash;
+  };
+};
+
+type SpaceRequestCancelled = {
+  type: "space_request_cancelled";
+  data: {
+    requestId: Hash;
+  };
+};
+
+type SpaceRequestAcceptanceRevoked = {
+  type: "space_request_acceptance_revoked";
+  data: {
+    requestAcceptanceId: Hash;
+  };
+};
+
+type ResourceRequested = {
+  type: "resource_requested";
+  data: {
+    resourceId: Hash;
+    eventId: Hash;
+    message: string;
+    timeSpan: TimeSpan;
+  };
+};
+
+type ResourceRequestAccepted = {
+  type: "resource_request_accepted";
+  data: {
+    requestId: Hash;
+  };
+};
+
+type ResourceRequestRejected = {
+  type: "resource_request_rejected";
+  data: {
+    requestId: Hash;
+  };
+};
+
+type ResourceRequestCancelled = {
+  type: "resource_request_cancelled";
+  data: {
+    requestId: Hash;
+  };
+};
+
+type ResourceRequestAcceptanceRevoked = {
+  type: "resource_request_acceptance_revoked";
+  data: {
+    requestAcceptanceId: Hash;
+  };
+};
+
+/**
+ * Roles
+ */
+
+type UserRoleAssigned = {
```

Comment:
Heavily under-specified, but there will be some way to assign roles to users....

Reply:
Looking good for now I think!